### DISCUSSION TITLE
The functionality of the proposed revealshape generated index

### DISCUSSION THREAD

**Discussion by jsbien** [rmast/revealshapes#3] [General]
Here is a fragment of my paper in print:

 What is sufficient for our purposes is
\textit{glyph} or \textit{character} \textit{spotting}. These tasks
are discussed in some publications, but there seems to be no tool
available directly for use. With some limitations, a variant of a
workflow described earlier can be used for this purpose. The
\textsf{djvudict} output can be converted to a \PDF\ document (created
with \TeX{}) with enlarged glyphs which make it relatively easy to
search for interesting items and to note their identifiers
(Fig.~\ref{fig:page}). Additionally, a \textsf{djview4poliqarp}
index can be created, which uses the shape identifier as the
searchable entry fields (Fig. \ref{fig:cis1}).  The identifiers are
not unique, nevertheless it is possible with some effort to find the
context of a shape in the document, as illustrated in Figures
\ref{fig:page} and \ref{fig:cis1} (note the shape
\texttt{01344}).
![page_017_10-crop](https://github.com/rmast/revealshapes/assets/8024702/91485001-31b6-4093-82cb-36dcc66fde57)

![cis](https://github.com/rmast/revealshapes/assets/8024702/2affdc36-6465-43de-b221-c3e3f04e975a)

The order of glyphs in the djvudict based PDF and index is, from the user point of view, practically random; this makes glyph spotting quite cumbersome. The proposed revealshapes index with the unique entries in the form `<file-name> d-<blit number>-<blit shapeno>` will allow, unless I'm mistaken, to group the similar shapes together just by sorting the entries.

In djview4shapes the shapes are for me too small to be browsed conveniently. Moreover their occurences on images are difficult to notice; this can be circumvented by opening them in djview4 and zooming the image but it is also not convenient. Nevertheless organizing the shapes into trees is useful and I will still occasionally use it despite its drawbacks.

BTW, I have some additional ideas about revealshapes, but it seems too early to talk about them :-)

**Comment by rmast** [comment]
I saw djview-poliqarp select a complete paragraph surrounded with the select-box, in which several found words (so probably index-lines) were covered. Does it enclose more search results into one selected region?

Verzonden vanaf Outlook voor Android<https://aka.ms/AAb9ysg>
________________________________
From: Janusz S. Bień ***@***.***>
Sent: Tuesday, April 16, 2024 6:11:54 AM
To: rmast/revealshapes ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [rmast/revealshapes] The functionality of the proposed revealshape generated index (Discussion #3)


Here is a fragment of my paper in print:

What is sufficient for our purposes is
\textit{glyph} or \textit{character} \textit{spotting}. These tasks
are discussed in some publications, but there seems to be no tool
available directly for use. With some limitations, a variant of a
workflow described earlier can be used for this purpose. The
\textsf{djvudict} output can be converted to a \PDF\ document (created
with \TeX{}) with enlarged glyphs which make it relatively easy to
search for interesting items and to note their identifiers
(Fig.~\ref{fig:page}). Additionally, a \textsf{djview4poliqarp}
index can be created, which uses the shape identifier as the
searchable entry fields (Fig. \ref{fig:cis1}). The identifiers are
not unique, nevertheless it is possible with some effort to find the
context of a shape in the document, as illustrated in Figures
\ref{fig:page} and \ref{fig:cis1} (note the shape
\texttt{01344}).
page_017_10-crop.png (view on web)<https://github.com/rmast/revealshapes/assets/8024702/91485001-31b6-4093-82cb-36dcc66fde57>

cis.png (view on web)<https://github.com/rmast/revealshapes/assets/8024702/2affdc36-6465-43de-b221-c3e3f04e975a>

The order of glyphs in the djvudict based PDF and index is, from the user point of view, practically random; this makes glyph spotting quite cumbersome. The proposed revealshapes index with the unique entries in the form d-- will allow, unless I'm mistaken, to group the similar shapes together just by sorting the entries.

In djview4shapes the shapes are for me too small to be browsed conveniently. Moreover their occurences on images are difficult to notice; this can be circumvented by opening them in djview4 and zooming the image but it is also not convenient. Nevertheless organizing the shapes into trees is useful and I will still occasionally use it despite its drawbacks.

BTW, I have some additional ideas about revealshapes, but it seems too early to talk about them :-)

**Comment by jsbien** [comment]
You refer I guess to the IMPACT corpus on the Poliqarp/marasca server. In this thread I would prefer to discuss only local *.csv index.

Nevertheless here is the reason you see what you see. The poliqarp index was created from XCES files created from the PAGE (I hate this misleading name) files created by hand by a Vietnamese firm paid by the IMPACT project. To save the money they were ordered to mark only the coordinates of the paragraphs. This is what we call 1-dimensional corpus. A quick-and-dirty script was written to get approximate coordinates for the lines. This we call 2-dimensional corpus.

As the result Poliqarp for a specific word in those corpora knows only the coordinates of, respectively, the paragraph or the line. It has nothing to do with the *.csv local indexes, which are created by a completely different workflow. One possible workflow was described in https://tug.org/TUGboat/tb44-3/tb138bien-rubricella.html (at the moment behind a paywall, but soon - some weeks I think - freely available).
